packages/core/src/external/carequality/ihe-gateway-v2/ihe-gateway-v2-logic.ts

```diff
@@ -206,22 +199,25 @@ export async function createSignSendProcessDrRequests({
     samlCertsAndKeys,
   });
 
-  const trustedKeyStore = await getTrustedKeyStore();
-
   const resultPromises = signedRequests.map(async (signedRequest, index) => {
     return sendProcessRetryDrRequest({
       signedRequest,
       samlCertsAndKeys,
       patientId,
       cxId,
       index,
-      trustedKeyStore,
     });
   });
 
-  const results = await Promise.all(resultPromises);
+  const results = await Promise.allSettled(resultPromises);
+  const successfulResults = results
+    .filter(
+      (result): result is PromiseFulfilledResult<OutboundDocumentRetrievalResp> =>
+        result.status === "fulfilled"
+    )
+    .map(result => result.value);
 
-  for (const result of results) {
+  for (const result of successfulResults) {
```

Comment:
We don't need to send the failed ones to the OSS Api?

Reply:
Shoot, ended up creating a new thread. Original one for reference: [link](https://github.com/metriport/metriport/pull/2268#discussion_r1643342639).